pages/profile.js

```diff
@@ -0,0 +1,26 @@
+import Link from 'next/link';
+import { supabase } from '../utils/initSupabase';
+
+export default function Profile({ user }) {
+  return (
+    <>
+      <Link href="/">
+        <a>Index</a>
+      </Link>
+      <pre>{JSON.stringify(user, null, 2)}</pre>
+    </>
+  );
+}
+
+export async function getServerSideProps({ req }) {
+  const token = req.cookies.supabase_auth_token;
+  const { data: user } = await supabase.auth.api.getUser(token);
+
+  if (!user) {
+    // If no user, redirect to index.
+    return { props: {}, redirect: { destination: '/', permanent: false } };
```

Comment:
Does this work? I know there's `res.redirect` in API routes but I'm not sure if redirect support landed for gSSP.

Reply:
It works fine for me. The docs say
> redirect - An optional redirect value to allow redirecting to internal and external resources. It should match the shape of { destination: string, permanent: boolean }`

https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering